src/lib.rs

```diff
@@ -138,7 +139,7 @@ pub struct Litep2p {
     local_peer_id: PeerId,
 
     /// Listen addresses.
-    listen_addresses: Vec<Multiaddr>,
+    listen_addresses: PublicAddresses,
```

Comment:
I don't think this is quite correct. Public addresses may be completely different form the listen addresses if we are behind NAT. We will also receive an error when try to bind to a public NAT address.

I would keep a clear separation between existing listen addresses and public addresses introduced in this PR.

Reply:
Yep that makes sense, I've introduced a similar object for ListenAddresses as for PublicAddresses, let me know what you think of it 🙏